File: 283. Move Zeroes/solution.cpp
// two pointer
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nonZeroIndex = 0;
        for(int zeroIndex = 0; zeroIndex<nums.size(); zeroIndex++){
            if(nums[zeroIndex] != 0){
                nums[nonZeroIndex] = nums[zeroIndex];
                nonZeroIndex++;
            }
        }
        for(nonZeroIndex; nonZeroIndex<nums.size(); nonZeroIndex++){
            nums[nonZeroIndex] = 0;
        }
    }
};

File: 283. Move Zeroes/solution.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        nonZeroIndex = 0;
        for zeroIndex in range(len(nums)):
            if(nums[zeroIndex] != 0):
                nums[nonZeroIndex] = nums[zeroIndex]
                nonZeroIndex += 1
            
        for nonZeroIndex in range (nonZeroIndex,len(nums)):
            nums[nonZeroIndex] = 0
